waterspout_api/serializers.py

```diff
@@ -78,6 +82,82 @@ def update(self, instance, validated_data):
 		return instance
 
 
+class EmailSerializer(serializers.Serializer):
+	email = serializers.EmailField()
+
+	class Meta:
+		fields = ("email",)
+
+
+class ResetPasswordSerializer(serializers.Serializer):
+	password = serializers.CharField(
+		write_only=True
+	)
+
+	class Meta:
+		fields = ("password",)
+
+	def validate(self, data):
+
+		# user's new password
+		password = data.get("password")
+		kwargs = self.context.get("kwargs", {})
+		token = kwargs.get("token")
+		encoded_pk = kwargs.get("encoded_pk")
+
+		if token is None:
+			raise serializers.ValidationError("Missing token")
+		elif encoded_pk is None:
+			raise serializers.ValidationError("Missing encoded_pk")
+
+		pk = urlsafe_base64_decode(encoded_pk).decode()
+
+		user = get_user_model().objects.get(pk=pk)
+		if not user:
+			raise serializers.ValidationError("User PK not found")
+
+		if PasswordResetTokenGenerator().check_token(user, token):
```

Comment:
My gut says it'd be good to move this functionality into the view, rather than the serializer. Were you seeing examples that had it here? I think of the serializer as validating the data is correct and ready to use. You could even check the token and raise the errror here and assume that if it returns, the user password can then be changed in the view.

But I could be completely mistaken on this - if you think it's best handled here, then leave it.